packages/engine/lib/error/src/frame.rs

```diff
@@ -14,37 +14,16 @@ use core::{
 use crate::provider::{self, Demand, Provider};
 use crate::Context;
 
-/// Trait alias to require all required traits used on a [`Frame`].
-///
-/// In order to implement these traits on a [`Frame`], the underlying type requires these types as
-/// well. After creation of the [`Frame`] it's erased. To unerase it later on to act on the actual
-/// Frame implementation, this trait is used.
-#[cfg(nightly)]
-trait Unerased: Provider + fmt::Debug + fmt::Display + Send + Sync + 'static {}
-#[cfg(nightly)]
-impl<T> Unerased for T where T: Provider + fmt::Debug + fmt::Display + Send + Sync + 'static {}
-
-/// Trait alias to require all required traits used on a [`Frame`].
-///
-/// In order to implement these traits on a [`Frame`], the underlying type requires these types as
-/// well. After creation of the [`Frame`] it's erased. To unerase it later on to act on the actual
-/// Frame implementation, this trait is used.
-#[cfg(not(nightly))]
-trait Unerased: fmt::Debug + fmt::Display + Send + Sync + 'static {}
-#[cfg(not(nightly))]
-impl<T> Unerased for T where T: fmt::Debug + fmt::Display + Send + Sync + 'static {}
-
-/// A single error, contextual message, or error context inside of a [`Report`].
+/// A single context or attachment inside of a [`Report`].
 ///
 /// `Frame`s are organized as a singly linked list, which can be iterated by calling
-/// [`Report::frames()`]. The head is pointing to the most recent context or contextual message,
-/// the tail is the root error created by [`Report::from_context()`] or [`Report::from_error()`].
-/// The next `Frame` can be accessed by requesting it by calling [`Report::request_ref()`].
+/// [`Report::frames()`]. The head is pointing to the most recent context or attachment, the tail is
```

Comment:
Does the head point to the most recent context or attachment? Or _is_ the head the most recent context / attachment?

Reply:
I clarified this in 35ecbb18150e064fc20dd7e2a10bd092437ec68e

Reply:
It's still somewhat confusing. 
* The head _is pointing to the_ current context or attachment
* The tail _is the_ root context

Do they behave different?

Reply:
Agree, this is confusing. 

Changed it to
>  The head contains the current context or attachment, and the tail contains the root context created by `Report::new()`.